dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -211,6 +284,43 @@ def is_match(self) -> bool:
             is_match = True
         return is_match
 
+    def _check_stop_condition_is_met(self, sample_size: int, unqiue_ratio: float):
+        """Return boolean given stop conditions.
+
+        :param sample_size: Number of samples to check the stop condition
+        :type sample_size: int
+        :param unqiue_ratio: Ratio of unique values to full sample size to
+            check stop condition
+        :type unqiue_ratio: float
+        :return: boolean for stop conditions
+        """
+        if (
+            self.max_sample_size_to_check_stop_condition is not None
+            and self.stop_condition_unique_value_ratio is not None
+            and sample_size >= self.max_sample_size_to_check_stop_condition
+            and unqiue_ratio >= self.stop_condition_unique_value_ratio
+        ):
+            return True
+        return False
+
+    def update_stop_condition(self, data: DataFrame):
```

Comment:
if we want this to continue to exist, I suggest making it private as we wouldn't want an external user to use this method.

Reply:
why wouldnt we want that?